scintillometry/psrio/psrfits_io.py

```diff
@@ -0,0 +1,383 @@
+                   'shape', 'sample_rate')
+
+    _header_defaults = main_header
+
+    def __init__(self, header_hdu=None):
+        if header_hdu is None:
+            super(PsrfitsHearderHDU, self).__init__()
+            self._init_empty()
+        else:
+            super(PsrfitsHearderHDU, self).__init__(header=header_hdu.header,
+                                                    data=header_hdu.data)
+            self.verify()
+
+    def verify(self):
+        if 'SIMPLE' not in list(self.header.keys()):
+            raise ValueError("The input HDU is not a fits headers HDU.")
+        if self.header['FITSTYPE'] != "PSRFITS":
+            raise ValueError("The input fits header is not a PSRFITS type.")
+
+    def _init_empty(self):
+        for k, v in self._header_defaults.items():
+            self.header.set(k, v['value'], v['comment'])
+        self.header.add_comment('FITS (Flexible Image Transport System) format'
+                                'is defined in Astronomy and Astrophysics, '
+                                'volume 376, page 359; bibcode:'
+                                ' 2001A&A...376..359H', after='EXTEND')
+
+    @property
+    def start_time(self):
+        # TODO add location
+        MJD_d_int = np.longdouble(self.header['STT_IMJD']) * u.day
+        MJD_s_int = np.longdouble(self.header['STT_SMJD']) * u.s
+        MJD_s_frac = np.longdouble(self.header['STT_OFFS']) * u.s
+        #NOTE I am assuming PSRFITS's time scale is UTC
+        return Time(MJD_d_int, MJD_s_int + MJD_s_frac, format='mjd',
+                    scale='utc', precision=9)
+
+    @start_time.setter
+    def start_time(self, time):
+        # TODO add location
+        if not isinstance(time, Time):
+            raise ValueError("Input time should be an `astropy.time.Time`"
+                             " object.")
+        if not time.isscalar:
+            raise ValueError("Input time should be a scalar")
+        else:
+            self.header['STT_IMJD'] = int(divmod(time.mjd, 1)[0])
+            day_int = Time(self.header['STT_IMJD'], scale=time.scale,
+                           format=time.format)
+            dt = (time - day_int).to(u.s)
+            int_sec, frac_sec = divmod(dt.value, 1)
+            self.header['STT_SMJD'] = int(int_sec)
+            self.header['STT_OFFS'] = frac_sec
+            self.header['DATE-OBS'] = time.fits
+
+    @property
+    def shape(self):
+        return (0,)
+
+    @property
+    def sample_rate(self):
+        return 0 * u.Hz
+
+    @property
+    def observatory(self):
+        return self.header['TELESCOP']
+
+    @observatory.setter
+    def observatory(self, val):
+        self.header['TELESCOP'] = val
+        #TODO ADD ITRF coordinate.
+
+    @property
+    def frequency(self):
+        try:
+            n_chan = float(self.header['OBSNCHAN'])
+            c_chan = float(self.header['OBSFREQ'])
+            bw = float(self.header['OBSBW'])
+        except:
```

Comment:
Usually best to enumerate actual exceptions that might happen, so that others are still raised. Definitely at least do `except Exception` so that a possible interrupt is not caught.